azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachineScaleSet.java

```diff
@@ -0,0 +1,1042 @@
+            DefinitionStages.WithInternalInternalLoadBalancerNatPool,
+            DefinitionStages.WithOS,
+            DefinitionStages.WithAdminUserName,
+            DefinitionStages.WithRootUserName,
+            DefinitionStages.WithLinuxCreate,
+            DefinitionStages.WithWindowsCreate,
+            DefinitionStages.WithCreate {
+    }
+
+    /**
+     * Grouping of virtual machine scale set definition stages.
+     */
+    @LangDefinition(ContainerName = "~/VirtualMachineScaleSet.Definition", ContainerFileName = "IDefinition", IsContainerOnly = true)
+    interface DefinitionStages {
+        /**
+         * The first stage of a virtual machine scale set definition.
+         */
+        interface Blank
+                extends GroupableResource.DefinitionWithRegion<VirtualMachineScaleSet.DefinitionStages.WithGroup> {
+        }
+
+        /**
+         * The stage of the virtual machine scale set definition allowing to specify the resource group.
+         */
+        interface WithGroup
+                extends GroupableResource.DefinitionStages.WithGroup<WithSku> {
+        }
+
+        /**
+         * The stage of the virtual machine scale set definition allowing to specify Sku for the virtual machines.
+         */
+        interface WithSku {
+            /**
+             * Specifies sku for the virtual machines in the scale set.
+             *
+             * @param skuType the sku type
+             * @return the stage representing creatable VM scale set definition
+             */
+            WithNetworkSubnet withSku(VirtualMachineScaleSetSkuTypes skuType);
+
+            /**
+             * Specifies sku for the virtual machines in the scale set.
+             *
+             * @param sku a sku from the list of available sizes for the virtual machines in this scale set
+             * @return the stage representing creatable VM scale set definition
+             */
+            WithNetworkSubnet withSku(VirtualMachineScaleSetSku sku);
+        }
+
+        /**
+         * The stage of the virtual machine scale set definition allowing to specify virtual network subnet for the
+         * primary network configuration.
+         */
+        interface WithNetworkSubnet {
+            /**
+             * Associate an existing virtual network subnet with the primary network interface of the virtual machines
+             * in the scale set.
+             *
+             * @param network an existing virtual network
+             * @param subnetName the subnet name
+             * @return the next stage of the virtual machine scale set definition
+             */
+            WithPrimaryInternetFacingLoadBalancer withExistingPrimaryNetworkSubnet(Network network, String subnetName);
+        }
+
+        /**
+         * The stage of the virtual machine scale set definition allowing to specify a public load balancer for
+         * the primary network interface of the scale set virtual machines.
```

Comment:
may be cleaner to say "[...] for the primary network interfaces of the virtual machines in the scale set."